src/main/java/org/cactoos/text/TextOf.java

```diff
@@ -342,15 +340,7 @@ public TextOf(final Iterable<?> iterable) {
      * @param iterator The iterable to convert to string
      */
     public TextOf(final Iterator<Character> iterator) {
-        this(
-            new TextOfScalar(
-                () -> {
-                    final StringBuilder buf = new StringBuilder();
-                    iterator.forEachRemaining(buf::append);
-                    return buf.toString();
-                }
-            )
-        );
+        this(new IterableOf<>(iterator));
```

Comment:
@DmitryBarskov please move this constructor above (good practice is to always have the called constructor/methods below the calling ones)